packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -2835,7 +2889,29 @@ class TextSelectionGestureDetector extends StatefulWidget {
 }
 
 class _TextSelectionGestureDetectorState extends State<TextSelectionGestureDetector> {
-  static int? _getDefaultMaxConsecutiveTap() => 2;
+  static int? _getDefaultMaxConsecutiveTap() {
+    switch(defaultTargetPlatform) {
+      // From observation, these platforms reset their tap count to 0 when the number of consecutive taps
+      // exceeds 3. For example on Linux, when going past a triple click, on the fourth click the selection
+      // is moved to the precise click position, on the fifth click the word at the position is selected, and
+      // on the sixth click the paragraph at the position is selected.
```

Comment:
Nit: Also maybe mention the distro and version of Linux since maybe they aren't all the same.